### DISCUSSION TITLE
How to get Signature field to render on form using custom markup?

### DISCUSSION THREAD

**Discussion by paulgoodfield** [solspace/craft-freeform#102] [Questions & Answers] [answered]
I have a form which includes a Signature field. I'm passing the form object to a custom template where I'm extracting all the field info etc so I can create custom markup for them.

As per the docs, for the Signature field I'm just using `{{ field.render() }}` to output it on the page.

At the bottom of the page (just before the `</body>` tag) I'm running `{{ craft.freeform.loadFreeformScripts(formObject) }}` to load the relevant scripts for the form.

However, the only script that gets loaded is the freeform.js file. The Signature field just shows as a blank space.

When I render the form using `{{ craft.freeform.form("myForm").render() }}` the Signature field renders successfully, but these additional scripts also get loaded:

- //unpkg.com/imask
- //cdn.jsdelivr.net/npm/signature_pad@2.3.2/dist/signature_pad.min.js

Is this expected behaviour? When I run `{{ craft.freeform.loadFreeformScripts(formObject) }}`, shouldn't it also load in the scripts needed to get the Signature field working?

**Comment by gustavs-gutmanis** [comment]
@paulgoodfield which version of Freeform are you on?
It should have worked the way you have it.  
I'll try and reproduce the issue and see what's up.

**Comment by paulgoodfield** [comment]
Apologies, should have mentioned I'm using version 3.10.6

**Comment by gustavs-gutmanis** [comment, marked as answer]
Sorry for the delay @paulgoodfield,

I tried out your scenario and the extra scripts don't get loaded if the form opening and closing tags aren't rendered by the form object.  
Here's the way I got it working:

```twig
{# Get the Form object #}
{% set form = craft.freeform.form('mySignatureForm') %}

{# Render the form's opening tag #}
{{ form.renderTag }}

{# Render the signature field#}
{{ form.get('signature').render }}
                
{# Render the form's closing tag #}
{{ form.renderClosingTag }}

..
..
..
..

{# Load the Freeform scripts #}
{{ craft.freeform.loadFreeformScripts(form) }}
</body>
```

Using Freeform to render both the `opening` and `closing` tags is crucial, because they trigger events which let various features inject their functionality into freeform.  
You can also just use the *load scripts inside form* feature in the Freeform's settings, because you can still do the custom markup, just remember to do the opening and closing tags via Freeform's functions.

**Reply by paulgoodfield** [reply]
Many thanks! Will try this out and get back to you if there are any issues.
